lib/buffer.js

```diff
@@ -782,6 +783,10 @@ function slowIndexOf(buffer, val, byteOffset, encoding, dir) {
       case 'utf-16le':
       case 'latin1':
       case 'binary':
+        if (!isValidEncoding(val, encoding)) {
```

Comment:
I am pretty sure you can replace this with something like:

```js
const { normalizeEncoding } = require('internal/util');

function isValidEncoding(val, encoding) {
  return !(normalizeEncoding(encoding) === 'hex' && val.length %2 == 0);
}
```

so we don't even need to call into C++ for this check.

Reply:
Actually, `isValidEncoding` is redundant here since `StringBytes::IsValidString` only checks for hex.

Reply:
Regarding replacing `isValidEncoding` in `crypto/cipher.js` and `crypto/hash.js`, what would be the benefit of using a new JS check instead of the current C++ one?

Reply:
It is slower to call into C++ from JavaScript versus just calling JavaScript from JavaScript.

Reply:
Thanks. I replaced `isValidEncoding` with a JS check and removed `StringBytes::IsValidString` since it's now unused.